### ISSUE TITLE
Uninstalling a package does not remove it from shrinkwrap

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#648]
### Code to reproduce the issue:

```
pnpm i lodash
cat shrinkwrap
pnpm uninstall lodash
cat shrinkwrap.yaml

pnpm i lodash
cat shrinkwrap
# remove from package.json dependencies
pnpm i
cat shrinkwrap.yaml
```

### Expected behavior:

Should be removed from shrinkwrap.

### Actual behavior:

lodash still in shrinkwrap.

### Additional information:

 - `node -v` prints: 6.5
 - Windows, OS X, or Linux?: macOS

**Comment by zkochan** [comment]
If you just do `pnpm i lodash`, it should not be saved in `shrinkwrap.yaml`. If you do `pnpm i -S lodash` then this is a valid case.

The `pnpm i -S lodash` case will be fixed by #649

**Comment by vjpr** [comment]
> If you just do pnpm i lodash, it should not be saved in shrinkwrap.yaml.

Just tested, when I run `pnpm i lodash`, it is saved in the shrinkwrap as of 0.58.0.

---

Ok, for uninstalling, if I use `pnpm uninstall -S lodash` it removes it from shrinkwrap too - this was my mistake.

**Comment by zkochan** [comment]
> Just tested, when I run pnpm i lodash, it is saved in the shrinkwrap as of 0.58.0.

yep, I am fixing this as well in #649

**Comment by vjpr** [comment]
The second case I mentioned is still broken.

```
pnpm i -S lodash
# manually remove lodash from package.json
rm -rf node_modules/lodash
pnpm i
cat shrinkwrap.yaml
```

lodash is still in the shrinkwrap. It should have been removed by `pnpm install` as it is no longer present in `node_modules` or `package.json` dependencies.

**Comment by zkochan** [comment]
ok, thanks, I'll create a test for this one

**Comment by zkochan** [comment]
Fix landed in [v0.60.0](https://github.com/pnpm/pnpm/releases/tag/v0.60.0)
